### ISSUE TITLE
Parallel code crashes with 1 core

### ISSUE THREAD

**Issue by BertvanderVeen** [kaskr/adcomp#390]
#### Description:
Using parallel_accumulator, code runs with single or multiple cores for small examples. For larger examples, code crashes with 1 core but runs fine with multiple  (>1) cores. Compiled with "TMBad".

#### Reproducible Steps:
**.cpp**
```
#include <TMB.hpp>
template<class Type>
Type objective_function<Type>::operator() ()
{
  DATA_MATRIX(Y);
  DATA_MATRIX(X);
  PARAMETER_MATRIX(b);
  parallel_accumulator<Type> nll(this);
  matrix<Type> eta(Y.rows(),Y.cols());
  eta.setZero();
  eta += X*b;

  for (int i=0; i<Y.rows(); i++) {
    for (int j=0; j<Y.cols();j++){
      nll -= dnorm(Y(i,j), eta(i,j), Type(1), true);
    }
  }
    
  return nll;
}
```

**.R**
```
library(TMB)
compile("linreg.cpp", framework = "TMBad")
dyn.load(dynlib("linreg"))
set.seed(123)
n = 12360
p = 57
X =  matrix(rnorm(2*n),ncol=2)
data <- list(Y = X%*%matrix(rnorm(p*2),ncol=p)+matrix(rnorm(n*p),ncol=p), X=X)
parameters <- list(b=matrix(0,ncol=p,nrow=2))
# This works
TMB::openmp(2)
obj <- MakeADFun(data, parameters, DLL="linreg")
# This crashes
TMB::openmp(1)
obj <- MakeADFun(data, parameters, DLL="linreg")
```

#### TMB Version:
1.9.10

#### R Version:
4.3.2

#### Operating System:
Ubuntu 22.04.3

**Comment by kaskr** [comment]
@BertvanderVeen Thanks for the report! I can confirm the crash.

**Comment by kaskr** [comment]
Some preliminary debugging tells me that for certain matrix sizes (triggered by this example), Eigen will parallelize matrix multiply when the openmp flag is enabled. This parallization conflicts with TMB AD types. One can disable Eigen parallization by modifying the example like this:

```cpp
#define EIGEN_DONT_PARALLELIZE
#include <TMB.hpp>
```

It should be considered whether TMB should automatically set this flag, and under which conditions.

**Comment by BertvanderVeen** [comment]
Thanks @kaskr, I can confirm that this suggestion solves the issue on my machine. Although the following is probably part of your follow-up, are there negative side-effects of setting `#define EIGEN_DONT_PARALLELIZE`?

**Comment by kaskr** [comment]
> are there negative side-effects...

No, it seems safest to set the flag until this issue has been solved in TMB.

**Comment by kaskr** [comment]
@BertvanderVeen Thanks again for reporting this! Fixed in 1.9.11 on CRAN.
